Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -140,87 +329,91 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable, IBufferPr
         public PythonTuple shape {
             get {
                 CheckBuffer();
-                return _shape;
+                return PythonTuple.Make(_shape);
             }
         }
 
         public PythonTuple strides {
             get {
                 CheckBuffer();
-                return _buffer.Strides;
+                return PythonTuple.Make(_strides);
             }
         }
 
         public PythonTuple suboffsets {
             get {
                 CheckBuffer();
-                return _buffer.SubOffsets ?? PythonTuple.EMPTY;
+                return PythonTuple.EMPTY;
             }
         }
 
         public Bytes tobytes() {
             CheckBuffer();
-            if (_matchesBuffer && _step == 1) {
-                return _buffer.ToBytes(_start / _itemsize, _end / _itemsize);
+
+            if (_shape.Contains(0)) {
+                return Bytes.Empty;
             }
 
-            byte[] bytes = getByteRange(_start, numberOfElements() * _itemsize);
+            var buf = _buffer!.AsReadOnlySpan();
 
-            if (_step == 1) {
-                return Bytes.Make(bytes);
+            if (_isCContig) {
+                return Bytes.Make(buf.Slice(_offset, _numItems * _itemSize).ToArray());
             }
 
-            // getByteRange() doesn't care about our _step, so if we have one
-            // that isn't 1, we will need to get rid of any bytes we don't care
-            // about and potentially adjust for a reversed memoryview.
-            byte[] stridedBytes = new byte[bytes.Length / _itemsize];
-            for (int indexStrided = 0; indexStrided < stridedBytes.Length; indexStrided += _itemsize) {
+            byte[] bytes = new byte[_numItems * _itemSize];
+            copyDimension(buf, bytes, _offset, dim: 0);
+            return Bytes.Make(bytes);
 
-                int indexInBytes = indexStrided * _step;
+            int copyDimension(ReadOnlySpan<byte> source, Span<byte> dest, int ofs, int dim) {
+                if (dim >= _shape.Count) {
+                    // copy individual element (scalar)
+                    source.Slice(ofs, _itemSize).CopyTo(dest);
+                    return _itemSize;
+                }
 
-                for (int j = 0; j < _itemsize; j++) {
-                    stridedBytes[indexStrided + j] = bytes[indexInBytes + j];
+                int copied = 0;
+                for (int i = 0; i < _shape[dim]; i++) {
+                    copied += copyDimension(source, dest.Slice(copied), ofs, dim + 1);
+                    ofs += _strides[dim];
                 }
+                return copied;
             }
-
-            return Bytes.Make(stridedBytes);
         }
 
-        public PythonList tolist() {
+        public object tolist() {
```

Comment:
In what case would this not return a `PythonList`?

Reply:
when `ndim == 0`.

Reply:
Maybe I add a test for this case.